src/main/java/org/broadinstitute/dsm/cf/TestBostonKitTrackerDispatcher.java

```diff
@@ -0,0 +1,134 @@
+package org.broadinstitute.dsm.cf;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretPayload;
+import com.google.cloud.secretmanager.v1.SecretVersionName;
+import com.google.pubsub.v1.PubsubMessage;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import org.apache.commons.dbcp2.ConnectionFactory;
+import org.apache.commons.dbcp2.DriverManagerConnectionFactory;
+import org.apache.commons.dbcp2.PoolableConnection;
+import org.apache.commons.dbcp2.PoolableConnectionFactory;
+import org.apache.commons.dbcp2.PoolingDataSource;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
+
+/**
+ * Identifies kits whose shipping history should be
+ * queried (by another job) in order to produce
+ * reports on turnaround time and study adherence.
+ */
+public class TestBostonKitTrackerDispatcher implements BackgroundFunction<PubsubMessage> {
+
+    private static final Logger logger = Logger.getLogger(TestBostonKitTrackerDispatcher.class.getName());
+
+    private String STUDY_MANAGER_SCHEMA = System.getenv("STUDY_MANAGER_SCHEMA") + ".";
+    private String STUDY_SERVER_SCHEMA = System.getenv("STUDY_SERVER_SCHEMA") + ".";
+
+
+    @Override
+    public void accept(PubsubMessage pubsubMessage, Context context) throws Exception {
+        String projectId = System.getenv("PROJECT_ID");
+        String secretId = System.getenv("SECRET_ID");
+
+
+        String KIT_QUERY =
+                "select\n" +
+                "u.guid,\n" +
+                "k.kit_label,\n" +
+                "from_unixtime(dkr.created_date/1000) as kit_requested_at,\n" +
+                "k.tracking_to_id,\n" +
+                "k.tracking_return_id,\n" +
+                "json_extract(k.test_result, '$[0].result') test_result,\n" +
+                "ifnull(STR_TO_DATE(replace(json_extract(k.test_result, '$[0].timeCompleted'),'\"',''), '%Y-%m-%dT%H:%i:%sZ'),\n" +
+                "    STR_TO_DATE(replace(json_extract(k.test_result, '$[0].timeCompleted'),'\"',''), '%Y-%m-%dT%H:%i:%s.%fZ')\n" +
+                ")as test_completion_time,\n" +
+                "u.hruid,\n" +
+                "dkr.upload_reason\n" +
+                "from\n" +
+                STUDY_MANAGER_SCHEMA + "ddp_kit k,\n" +
```

Comment:
Making `KIT_QUERY` a static var may run into caching problems.  Having different schema names between dev and prod is great for minimizing the chance of accidents, but calling out the schemas differently here is a bit clunky.